frontend/src/metabase/admin/settings/components/widgets/GroupMappingsWidget.jsx

```diff
@@ -108,9 +197,39 @@ export default class GroupMappingsWidget extends React.Component {
     );
   };
 
+  updateGroupsFromDeletedMappings = () => {
+    const { groupsToDelete } = this.state.whenDeletingMappingGroups;
+
+    groupsToDelete.forEach(id => PermissionsApi.deleteGroup({ id }));
```

Comment:
Call API directly from event handlers is not a great idea, it can also lead to bugs:

<video src="https://user-images.githubusercontent.com/14301985/214121809-a6590546-e631-40ef-9b9e-f6a4147e0892.mov" />

Reply:
> I am wondering why we decided to go without inlining the form. 

@alxnddr let me try to recap the steps that made me go this way. 

It may be illustrative of how investing time in the transition from product doc to tickets (Github issues) can make us more productive, as our team talked about today in our retro.

### 1.

The issue that is on our tactical board, and is the gateway to this PR is https://github.com/metabase/metabase/issues/27212

Note how the issue description starts by saying "this is a small update".

Description is followed by links to the product doc and Figma. 
It may be worth noting that Figma is about the confirmation modal (what to do about groups), and afaik no Figma is about inlining the widget (understandable from one point-of-view, as it's intuitive what inlining would mean, but it brings the confirmation modal even more to the top as the main thing to tackle here).

Description next has "Implementation plan". 
This is what I went for first.
The implementation plan is what this PR does.
One thing that clearly needed to be tackled, and wasn't either in the implementation plan or in the product doc, was handling multiple groups per mapping, and edge cases around this fact. These were tackled here after discussions with Noah and Maz both in Slack and in Notion.

### 2.

After the above was taken care of, I went back to the product doc.

This PR was of a size that I already would veer towards doing further changes in a follow-up PR.

At any rate, I dedicated time to understanding different uses of the widget.
It became clear to me, and only at this point really, that this widget in SAML fits inside a narrow layout.
Also I learned that layouts for LDAP and JWT are different from the SAML layout.
Furthermore, the way the widget is called is a bit different in each of the three uses it gets.

I continued product discussions in our [Notion doc](https://www.notion.so/metabase/Make-SSO-group-mapping-clearer-b9ed33e4ee8c4facb2220f20b565e51f), see the top of the @today section.

And this discussion led to a decision by Maz and Arakaki on how to go about this.

Reply:
From what's been decided today about inlining, what do you think of merging this iteration which works, and next we inline the widget?

Reply:
@gusaiani got it, thanks for describing it comprehensively. It looks like we are slightly misaligned about what is the function of epics, issues, and product docs. This question has been raised a couple of times and as far as I remember we don't put full specifications with all the details in GitHub issues intentionally instead their role is more utilitarian and when necessary, providing additional info that can be valuable for engineers such as tech details, caveats. So before starting projects engineers need to read and understand product docs. If you think it makes sense to change the process and start putting the entire specs with all the details and mockups in the issues we need to align on that first. Something like an RFC on this process change would be helpful.

Regarding merging this approach as is, I think implementing the inline form right away would make more sense because we are going to ship the inline version and it has less logic. So the current implementation seems not like a first step towards the project implementation which would be fine to merge, but going in a different direction

Reply:
@alxnddr thanks.

I will continue this task by inlining the widget.
Will do this in a subbranch for now, just for a minor separation of diff states.